pkg/webhooks/machine_webhook.go

```diff
@@ -925,6 +931,25 @@ func validateAzure(m *machinev1beta1.Machine, config *admissionConfig) (bool, []
 			fmt.Sprintf("osDisk.diskSettings.ephemeralStorageLocation can either be omitted or set to %s", azureEphemeralStorageLocationLocal)))
 	}
 
+	if mOld != nil {
+		oldProviderSpec := new(machinev1beta1.AzureMachineProviderSpec)
+		if err := unmarshalInto(mOld, oldProviderSpec); err != nil {
+			errs = append(errs, err)
+			return false, warnings, errs
+		}
+		if providerSpec.CapacityReservationGroupID != "" && !validateAzureImmutabilityForCapacityReservationGroupID(oldProviderSpec.CapacityReservationGroupID, providerSpec.CapacityReservationGroupID) {
+			errs = append(errs, field.Invalid(field.NewPath("providerSpec", "capacityReservationGroupID"),
+				providerSpec.CapacityReservationGroupID, "capacityReservationGroupID is immutable"))
+		}
```

Comment:
Will this execute on only Machines or also MachineSets?

We don't typically validate immutability inside providerSpecs since the type is re-used across the Machines and MachineSets and I don't believe we distinguish between the two when this validation code is called

Reply:
@JoelSpeed This is used in both machine and machineset

Reply:
It shouldn't validate machinesets?

Reply:
So should we remove the immutability validation from everywhere?

Reply:
Only downstream, for this feature yes. Where else is it?

Reply:
it is written in https://github.com/openshift/api/pull/1866 in  godocs should be remove it from there as well.